### PR TITLE
#166 add enrollment key tab for lecturer

### LINKED ISSUE #166: As a lecturer I can display the key that is required to join a lecture (1)
AC:
- A screen exists where the key to join a lecture is displayed
- No other keys are displayed on this screen
- The screen is either
    - reachable from either a lecture or course view
    - directly on a lecture or course view
- I can get back to the screen I came from from this screen

### PR DESCRIPTION
**Pull request by lpfennigschmidt**
# Resolves #166 

With this PR we include a tab in the tab view for the lecturer to show the enrollment key to his/her students

### CODE REVIEW

**Comment by adrianjost** [comment]
@lpfennigschmidt Could you review this PR please: #121 ? This PR will also introduce some changes to the form and should make resolving the conflicts to the _form easier.

File: app/views/lectures/show.html.erb [hunk 1]
```diff
@@ -12,6 +12,9 @@
     <a class="nav-link" id="feedback-tab" data-toggle="tab" href="#feedback" role="tab" aria-controls="feedback" aria-selected="false">Feedback</a>
   </li>
   <% if !@current_user.is_student %>
+    <li class="nav-item">
+      <a class="nav-link" id="enrollmentKey-tab" data-toggle="tab" href="#enrollmentKey" role="tab" aria-controls="enrollmentKey" aria-selected="false">Enrollment Key</a>
+    </li>
```
Comment (grittaweisheit) [inline app/views/lectures/show.html.erb]: Do we really want to have the tab even if we don't have a key for the lecture?

File: app/views/lectures/show.html.erb [hunk 2]
```diff
@@ -73,8 +76,11 @@
 
   </div>
   <% if !@current_user.is_student %>
+    <div class="tab-pane fade" id="enrollmentKey" role="tabpanel" aria-labelledby="enrollmentKey-tab-tab">
+      <h1 class="padded-text">Enrollment Key: <%= @lecture.enrollment_key %></h1>
+    </div>
```
Comment (grittaweisheit) [inline app/views/lectures/show.html.erb]: Again: what if there is no Key? Maybe a shortcut for adding a key would be nice in that case :)

File: app/assets/stylesheets/lectures.scss [hunk 3]
```diff
@@ -18,4 +18,10 @@ form.button_to {
 
 .tab-pane {
   padding-top: 20px;
+}
+
+.padded-text {
+  padding-top: 20px;
+  padding-bottom: 20px;
+  padding-left: 20px;
```
[the remark this answers is not in the document]
Comment (adrianjost) [reply app/assets/stylesheets/lectures.scss]: or use
```css
.padding-text{
  padding: 20px;
  padding-right: 0;
}
```

File: app/views/lectures/show.html.erb (hunk 1 above)
Comment (adrianjost) [reply app/views/lectures/show.html.erb]: keys are currently required

Comment (grittaweisheit) [reply app/views/lectures/show.html.erb]: Ah, they shouldn't be according to #164 . Sorry, I assumed it was in this story, too, though it isn't.

**Pull request merged**